src/VisualStudio/Core/Def/Implementation/VisualStudioCodeDefinitionWindowService.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Editor;
+using System;
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.VisualStudio.Shell.Interop;
+using System.Composition;
+using System.Runtime.InteropServices;
+using SVsServiceProvider = Microsoft.VisualStudio.Shell.SVsServiceProvider;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [Export(typeof(ICodeDefinitionWindowService)), Shared]
+    internal sealed class VisualStudioCodeDefinitionWindowService : ICodeDefinitionWindowService
+    {
+        private readonly IVsCodeDefView _vsCodeDefView;
+
+        [ImportingConstructor]
+        public VisualStudioCodeDefinitionWindowService(SVsServiceProvider serviceProvider)
+        {
+            _vsCodeDefView = (IVsCodeDefView)serviceProvider.GetService(typeof(SVsCodeDefView));
+        }
+
+        public void SetContext(ImmutableArray<CodeDefinitionWindowLocation> locations)
+        {
+            // If the new context has no location, then just don't update, instead of showing the
```

Comment:
Oh that's the intent? How about two methods, SetContext (must be non-empty list) and Clear?